salt/utils/__init__.py

```diff
@@ -1920,3 +1921,45 @@ def output_profile(pr, stats_path='/tmp/stats', stop=False, id_=None):
         stacklevel=3
     )
     return salt.utils.profile.output_profile(pr, stats_path, stop, id_)
+
+
+def get_module_environment(env=None):
+    '''
+    Get module optional environment.
+
+    To setup an environment option for a particular module,
+    add either pillar or config at the minion as follows:
+
+    system-environment:
+      salt.modules.pkg:
+        LC_ALL: en_GB.UTF-8
+        FOO: bar
+      salt.states.pkg:
+        LC_ALL: en_US.Latin-1
+        NAME: Fred
```

Comment:
I'm not sure I like this structure. A lot of users aren't developers, and will thus not intrinsically grok the "module path" naming. From a user-friendliness standpoint, I think it would be more intuitive to do something like:

```yaml
system-environment:
  modules:
    pkg:
      FOO: bar
  states:
    pkg:
      HELLO: world
```

@cachedout thoughts on this?

Reply:
@cachedout please agree with the @terminalmage because I like his proposal more than mine. :wink: